src/arch/x86/pagetable_walker.cc

```diff
@@ -511,7 +511,9 @@ Walker::WalkerState::stepWalk(PacketPtr &write)
     if (doEndWalk) {
         if (doTLBInsert)
             if (!functional)
-                walker->tlb->insert(entry.vaddr, entry);
+                //This is needed to just compile gem5 (since the tlb->insert()
+                //is updated), but would not effect SE mode runs
+                walker->tlb->insert(entry.vaddr, entry, 0);
```

Comment:
Do we have access to the ThreadContext here? If we were to use this fix for FS mode, do we know what the ASID would be?

Reply:
This needs to be addressed. Why always 0?

It doesn't look like we have access to the thread context here. So, we aren't going to be able to support multiple hardware threads in FS mode. This is OK, but we need to put a panic/fatal somewhere to throw an error if we try.

Reply:
Initially, I was not sure if we have access to the thread context here (so kept the last parameter to 0, just to compile and see if things work for SE mode). Looking more into it, I observed that walker state actually has access to ThreadContext, so updated to use that in a new commit